Planit/View/Utils/CustomTextEditor.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  CustomTextEditor.swift
+//  Planit
+//
+//  Created by Shivangi on 06/12/22.
+//
+
+import SwiftUI
+
+struct CustomTextEditor: View{
+    
+    
+    @Binding var text: String
+    let placeholder: Text
+    let imageName: String
+    var foregroundColor: Color?
```

Comment:
I would prefer to make this a non-optional and then you only have to deal with the "fallback" color once when you set it. Now you have to check for the optional value more often.

Use optionals when you need to, but try to use non-optionals whenever possible!

Instead of the `if let safeForegroundColor`, I would do it like `self.foregroundColor = foregroundColor ?? .white` in the initializer. :)